data/src/main/java/ca/bc/gov/data/datasource/remote/HospitalVisitRemoteDataSource.kt

```diff
@@ -0,0 +1,29 @@
+package ca.bc.gov.data.datasource.remote
+
+import ca.bc.gov.common.BuildConfig
+import ca.bc.gov.common.const.MESSAGE_INVALID_RESPONSE
+import ca.bc.gov.common.const.SERVER_ERROR
+import ca.bc.gov.common.exceptions.MyHealthException
+import ca.bc.gov.common.model.hospitalvisits.HospitalVisitDto
+import ca.bc.gov.data.datasource.remote.api.HealthGatewayPrivateApi
+import ca.bc.gov.data.model.mapper.toDto
+import ca.bc.gov.data.utils.safeCall
+import javax.inject.Inject
+
+class HospitalVisitRemoteDataSource @Inject constructor(
+    private val healthGatewayPrivateApi: HealthGatewayPrivateApi
+) {
+
+    suspend fun getHospitalVisit(token: String, hdid: String): List<HospitalVisitDto> {
+        if (BuildConfig.FLAG_HOSPITAL_VISITS.not()) return emptyList()
```

Comment:
we are checking for FLAG_HOSPITAL_VISITS while calling repository method. so do we require to check here again? if possible, add if condition for FLAG_HOSPITAL_VISITS only once instead of checking it everywhere.

Reply:
Thanks, I'm going to address in the next PR